cip/CIP2017-06-18-multiple-graphs.adoc

```diff
@@ -0,0 +1,476 @@
+INTO NEW GRAPH recommendations
+// containing existing nodes and new rels ...
+CREATE (a)-[:POSSIBLE_FRIEND]->(c)
+// ... and finally discard all tabular data and cardinality
+WITH GRAPHS *
+
+// Switch context to named graph.
+FROM GRAPH recommendations
+MATCH (a:Person)-[e:POSSIBLE_FRIEND]->(b:Person)
+// Return tabular and graph output
+RETURN a.name, b.name, count(e) AS cnt
+    ORDER BY cnt DESC
+    GRAPHS recommendations
+----
+
+Example 5
+[source, cypher]
+----
+// Set scope to the whole social network ...
+FROM GRAPH AT 'graph://social-network'
+// .. and match some data.
+MATCH (a:Person)-[:IS_LOCATED_IN]->(c:City),
+      (c)->[:IS_LOCATED_IN]->(co:Country),
+      (a)-[e:KNOWS]-(b)
+
+// Create a new temporary named graph,
+INTO NEW GRAPH sn_updated
+// add previous matches to new graph,
+CREATE (a)-[e]-(b)
+// update existing nodes.
+SET a.country = cn.name
+// ... and finally discard all tabular data and cardinality
+WITH GRAPHS *
+
+FROM GRAPH sn_updated
+MATCH (a:Person)-[e:KNOWS]->(b:Person)
+WITH a.country AS a_country, b.country AS b_country, count(a) AS a_cnt, count(b) AS b_cnt, count(e) AS e_cnt
+INTO NEW GRAPH rollup {
+  MERGE (:Persons {country: a_country, cnt: a_cnt})-[:KNOW {cnt: e_cnt}]->(:Persons {country: b_country, cnt: b_cnt})
+}
+// Return final graph output
+RETURN GRAPHS rollup
+----
+
+Example 5
+[source, cypher]
+----
+// Set scope to the whole social network ...
+FROM GRAPH AT 'graph://social-network'
+// .. and match some data.
+MATCH (a:Person)-[e]->(b:Person),
+      (a)-[:LIVES_IN]->()->[:IS_LOCATED_IN]-(c:Country {name: ‘Sweden’}),
+      (b)-[:LIVES_IN]->()->[:IS_LOCATED_IN]-(c)
+// Create a persistent graph at 'graph://social-network/swe'
+INTO NEW GRAPH sweden_people AT './swe' {
+  // connecting persons that live in the same city in Sweden.
+  CREATE (a)-[e]->(b)
+}
+// Finally discard all tabular data and cardinality
+WITH GRAPHS *
+
+MATCH (a:Person)-[e]->(b:Person),
+      (a)-[:LIVES_IN]->()->[:IS_LOCATED_IN]-(c:Country {name: ‘Germany’}),
+      (b)-[:LIVES_IN]->()->[:IS_LOCATED_IN]-(c)
+// Create a persistent graph at 'graph://social-network/ger'
+INTO NEW GRAPH german_people AT './ger' {
+  // connecting persons that live in the same city in Germany.
+  CREATE (a)-[e]->(b)
+}
+// Finally discard all tabular data and cardinality
+WITH GRAPHS *
+
+// Start query on the 'sweden_people' graph
+FROM GRAPH sweden_people
+MATCH p=(a)--(b)--(c)--(a) WHERE NOT (a)--(c)
+// Create a temporary graph 'swedish_triangles'
+INTO NEW GRAPH swedish_triangles {
+  ADD p
+}
+// and return it together with a count of it's content
+RETURN count(p) AS num_triangles GRAPHS swedish_triangles, sweden_people, german_people
+----
+
+== Interaction with existing features
+
+This proposal is far reaching as it changes both the property graph model and the execution model of the language.
+
+However, the change has been carefully designed to not change the semantics of existing queries.
+
+== Alternatives
```

Comment:
I think this and subsequent sections are superfluous since the introduction of CIRs. We should modify our template.